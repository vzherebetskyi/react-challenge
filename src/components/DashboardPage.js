import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { auth } from '../firebase/firebase';
import { startLogoutUser } from '../actions/auth';

const DashboardPage = () => {
  const dispatch = useDispatch();
  const userProfile = useSelector(state => state.userProf);

  const handleLogout = () => {
    dispatch(startLogoutUser(auth))();
  };

  return (
    <div className="page-container">
      <div className="page-wrapper" style={{ alignItems: 'center' }}>
        <div>
          <h1 className="base-title">Dashboard</h1>
          <div style={{ display: 'flex' }}>
            <h3 style={{ marginRight: '1rem' }}>Username:</h3>
            <h3>{userProfile ? userProfile.username : ''}</h3>
          </div>
          <div style={{ display: 'flex' }}>
            <h3 style={{ marginRight: '1rem' }}>User email:</h3>
            <h3>{userProfile ? userProfile.email : ''}</h3>
          </div>
          <div className="parall-styled-btn-container">
            <button
              style={{ marginTop: '2rem', left: 0 }}
              className="parall-styled-btn"
              onClick={handleLogout}
            >
              <div>Logout</div>
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DashboardPage;
