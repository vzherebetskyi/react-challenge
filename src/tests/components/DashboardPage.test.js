import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

import DashboardPage from '../../components/DashboardPage';

// afterEach(cleanup);

jest.mock('firebase/app', () => {
  return {
    initializeApp: jest.fn(),
  };
});

jest.mock('firebase/auth', () => {
  return {
    getAuth: jest.fn(),
  };
});

const mockDispatch = jest.fn(() => jest.fn());
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

test('should render DashboardPage correctly', () => {
  const elem = renderer.create(<DashboardPage />).toJSON();
  expect(elem).toMatchSnapshot();
});

test('dispatch is called on Log out button click', () => {
  const { container } = render(<DashboardPage />);
  const button = container.querySelector('button');
  fireEvent.click(button);
  expect(mockDispatch).toHaveBeenCalledTimes(1);
});
