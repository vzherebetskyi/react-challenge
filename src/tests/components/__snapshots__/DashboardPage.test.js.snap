// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
<div
  className="page-container"
>
  <div
    className="page-wrapper"
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <div>
      <h1
        className="base-title"
      >
        Dashboard
      </h1>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <h3
          style={
            Object {
              "marginRight": "1rem",
            }
          }
        >
          Username:
        </h3>
        <h3>
          
        </h3>
      </div>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <h3
          style={
            Object {
              "marginRight": "1rem",
            }
          }
        >
          User email:
        </h3>
        <h3>
          
        </h3>
      </div>
      <div
        className="parall-styled-btn-container"
      >
        <button
          className="parall-styled-btn"
          onClick={[Function]}
          style={
            Object {
              "left": 0,
              "marginTop": "2rem",
            }
          }
        >
          <div>
            Logout
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
